ocm_python_wrapper/cluster.py

```diff
@@ -257,33 +258,112 @@ def addon_info(self):
+
+        Args:
+            user_parameters (List) : User parameter input set with values.
+            use_api_defaults (bool) : If true, set required parameter with default value to not fail as missing.
+
+        Returns: Updated parameters (if default values updated) to provide for installation.
+
+        Raises:
+            ValueError: When a required parameter is missing,
+                        or when parameters are passed but not needed for addon.
+        """
+
+        def _get_required_cluster_parameters(parameters):
+            """Uses documentation of the addon parameters and filters them by conditions related to the cluster.
+
+            Args:
+                parameters (dict) : Addons parameters from api.
+
+            Returns:
+                Dict of parameters corresponding to the clusters the current configuration and
+                              required for installation.
+
+            Examples:
+                parameters = {
+                'items': [{'id': 'rosa-cli-required', 'name': 'ROSA CLI Required', 'description':
+            '...', 'value_type': 'boolean', 'validation': '^true$', 'validation_err_msg': '', 'required': True,
+            'editable': False, 'enabled': True, 'default_value': '', 'editable_direction': '', 'conditions': [{
+            'resource': 'cluster', 'data': {'aws.sts.enabled': True, 'cloud_provider.id': 'aws'}}]}, ...]
+            }
+
+                _required_parameters = {'cidr-range': {'default_value': '10.1.0.0/26'}, 'addon_parameter': {
+                        'default_value': ''}, ..}
+            """
+            _required_parameters = {}
+
+            for param in parameters["items"]:
+                if param["required"] is True:
+                    param_conditions = [
+                        con["data"]
+                        for con in param["conditions"]
+                        if con["resource"] == "cluster"
+                    ]
+                    if param_conditions:
+                        for condition, condition_value in param_conditions[0].items():
+                            cluster_condition_value = benedict(
+                                self.instance.to_dict(), keypath_separator="."
+                            ).get(condition)
+                            if not (
+                                (
+                                    isinstance(condition_value, list)
+                                    and cluster_condition_value in condition_value
+                                )
+                                or cluster_condition_value == condition_value
+                            ):
+                                break
+                        else:
+                            _required_parameters[param["id"]] = {
+                                "default_value": param["default_value"]
+                            }
+
+            return _required_parameters
+
+        addon_parameters = self.addon_info().get("parameters")
```

Comment:
If `addon_parameters` is None (i.e no `parameters` key - is that a valid option?), `required_parameters` functionality will fail